Libplanet/Action/Loader/TypedActionLoader.cs

```diff
@@ -0,0 +1,175 @@
+        /// </para>
+        /// </summary>
+        /// <param name="types">The <see cref="IValue"/>s to <see cref="Type"/>s mapping to use.
+        /// </param>
+        /// <seealso cref="TypedActionLoader.Create"/>
+        public TypedActionLoader(IDictionary<IValue, Type> types)
+        {
+            _types = types;
+        }
+
+        public IDictionary<IValue, Type> Types => _types;
+
+        /// <inheritdoc cref="IActionLoader.LoadAction"/>.
+        /// <summary>
+        /// <para>
+        /// Creates a new <see cref="TypedActionLoader"/> instance.
+        /// </para>
+        /// <para>
+        /// The <see cref="IAction"/> type to load is restricted to given
+        /// <paramref name="assembly"/> and <paramref name="baseType"/>,
+        /// except for those handled by <see cref="Registry"/> for system <see cref="IAction"/>s.
+        /// </para>
+        /// <para>
+        /// This automatically searches an <see cref="Assembly"/> with all <see cref="IAction"/>s
+        /// according to the following rule:
+        /// <list type="bullet">
+        ///     <item><description>
+        ///         The <see cref="Assembly"/> to search is <paramref name="assembly"/> if given,
+        ///         or the <see cref="Assembly"/> of <paramref name="baseType"/> if not.
+        ///     </description></item>
+        ///     <item><description>
+        ///         From the <see cref="Assembly"/> determined above, all <see cref="IAction"/>s
+        ///         with <see cref="ActionTypeAttribute"/> attached are collected.
+        ///     </description></item>
+        ///     <item><description>
+        ///         If <paramref name="baseType"/> is provided, only those <see cref="IAction"/>
+        ///         types that are assignable to <paramref name="baseType"/> are selected.
+        ///     </description></item>
+        /// </list>
+        /// </para>
+        /// <para>
+        /// Then a dictionary mapping from <see cref="IValue"/>s to <see cref="Type"/>s created
+        /// from searched <see cref="IAction"/>s and the explicit loader is created by
+        /// passing the map to <see cref="TypedActionLoader(IDictionary{IValue, Type})"/>.
+        /// </para>
+        /// </summary>
+        /// <param name="assembly">The <see cref="Assembly"/> to search <see cref="IAction"/> types.
+        /// </param>
+        /// <param name="baseType">The type of <see cref="IAction"/> to filter.</param>
+        /// <returns>A <see cref="TypedActionLoader"/> created from provided
+        /// <paramref name="assembly"/> and <paramref name="baseType"/>.</returns>
+        /// <exception cref="ArgumentException">Thrown when both <paramref name="assembly"/> and
+        /// <paramref name="baseType"/> are <see langword="null"/>.
+        /// </exception>
+        /// <seealso cref="TypedActionLoader(IDictionary{IValue, Type})"/>
+        public static TypedActionLoader Create(Assembly? assembly = null, Type? baseType = null)
```

Comment:
Even in `TypedActionLoader`, should we care about `baseType`?

Reply:
Here `baseType` is optional. This serves several purposes.

- `PolymorphicAction<T>` had to have **something** for `T`, and this allows to mimic the previous `PolymorphicAction<T>`'s behavior (before `StaticActionLoader` was introduced).
- Such behavior can still act as a defensive mechanism to filter types (just in case there are some unwanted attributed `IAction`s lying around in an assembly?).
- If unwanted, it can simply be ignored. 😗